makeflow/src/makeflow.c

```diff
@@ -806,19 +645,21 @@ static void makeflow_node_complete(struct dag *d, struct dag_node *n, struct bat
 		makeflow_log_state_change(d, n, DAG_NODE_STATE_FAILED);
 
 		/* Clean files created in node. Clean existing and expected and record deletion. */
-		list_first_item(n->target_files);
-		while((f = list_next_item(n->target_files))) {
+		list_first_item(outputs);
+		while((f = list_next_item(outputs))) {
 			if(f->state == DAG_FILE_STATE_EXPECT) {
-				makeflow_file_clean(d, remote_queue, f, 1);
+				makeflow_clean_file(d, remote_queue, f, 1);
 			} else {
-				makeflow_file_clean(d, remote_queue, f, 0);
+				makeflow_clean_file(d, remote_queue, f, 0);
 			}
 		}
 
-		if(monitor_mode && info->exit_code == 147)
+		if(monitor && info->exit_code == 147)
```

Comment:
Ah, I forgot about this ugliness. In the future we want exit_code = 0 if task did not overflow, -1 otherwise, and get the task status from information that the monitor generates.